src/validation_workflow/validation_tar.py

```diff
@@ -0,0 +1,28 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+
+from system.temporary_directory import TemporaryDirectory
+from validation_workflow.download_utils import DownloadUtils
+from validation_workflow.validation import Validation
+
+
+class ValidationTar(Validation, DownloadUtils):
+    tmp_dir = TemporaryDirectory()
+
+    @classmethod
+    def download_artifacts(self, projects: list, version: str, platform: str, architectures: list) -> bool:
+        for project in projects:
+            for architecture in architectures:
+                url = f"{self.base_url}{project}/{version}/{project}-{version}-{platform}-{architecture}.tar.gz"
+                if ValidationTar.is_url_valid(url) and ValidationTar.download(url, self.tmp_dir):
+                    logging.info(f" Valid URL - {url} and Download Successful !")
```

Comment:
All of these implementation are extremely similar, means the core code should exist in parent, and child should just pass the different information.

Reply:
I was initially suggested to have only abstract methods in the base class so I have implemented it this way. I can actually change the core code to base class and pass the required information from the child class. Can I know what you would recommend!

Reply:
I'm expecting to leave the method abstract in base class  as Docker  also extends the same method which has a different implementation all together.So, had to implement it independently for each distribution even though the code is reused.